modules/openapi-generator/src/main/resources/dart-dio/api.mustache

```diff
@@ -78,19 +78,22 @@ class {{classname}} {
             options: Options(
             method: '{{httpMethod}}'.toUpperCase(),
             headers: headerParams,
+            extra: {
+                'secure': [{{#authMethods}} {"type": "{{type}}", "name": "{{name}}"{{#isApiKey}}, "keyName": "{{keyParamName}}", "where": "{{#isKeyInQuery}}query{{/isKeyInQuery}}{{#isKeyInHeader}}header{{/isKeyInHeader}}"{{/isApiKey}} }{{#hasMore}}, {{/hasMore}}{{/authMethods}}],
+            },
             contentType: contentTypes.isNotEmpty ? contentTypes[0] : "application/json",
             ),
             cancelToken: cancelToken,
             ){{#returnType}}.then((response) {
 
         {{#isListContainer}}
                 final FullType type = const FullType(BuiltList, const [const FullType({{returnBaseType}})]);
-                BuiltList<{{returnBaseType}}> dataList = _serializers.deserialize(response.data, specifiedType: type);
+                BuiltList<{{returnBaseType}}> dataList = _serializers.deserialize(jsonDecode(response.data), specifiedType: type);
                 var data = dataList.toList();
         {{/isListContainer}}
         {{^isListContainer}}
         var serializer = _serializers.serializerForType({{{returnType}}});
-        var data = _serializers.deserializeWith<{{{returnType}}}>(serializer, response.data);
+        var data = _serializers.deserializeWith<{{{returnType}}}>(serializer, jsonDecode(response.data));
```

Comment:
What was the issue here? I've been using clients generated from this template without any deserialization issues.

Reply:
The issue was that out of the box nothing was working for me serializer was expecting a map and it wasn't happy with the string check #6026 for details. maybe I've done something wrong...

Reply:
Ah, I think I know why now. I use a Dio Transformer in all my clients: https://github.com/flutterchina/dio_flutter_transformer

I think a better approach would be to create default transformer for the generated client that parses the json. Then it can be overidden by a custom transformer.

We probably need to check what type the response is before deserializing and throw an error if its not a map.

Reply:
I'm super new to dio and don't know transformer ^^ but from what I read yes that's why you don't have any issue, I'll change my code to use a transformer then :)

Reply:
but from the look of the doc transformers are not always used :/ 
```
/// [transformer] allows changes to the request/response data before it is sent/received to/from the server
  /// This is only applicable for request methods 'PUT', 'POST', and 'PATCH'.
  Transformer transformer;
```

Reply:
why not check if response is string to use jsonDecode on it ?

Reply:
I think those docs are wrong because my GET requests are being transformed. 

Checking the response type is an ok approach as well.

Reply:
done